packages/grid/_modules_/grid/hooks/features/core/useGridApi.ts

```diff
@@ -1,13 +1,16 @@
+import * as React from 'react';
 import { ApiRef } from '../../../models/api/apiRef';
 import { GridApi } from '../../../models/api/gridApi';
 import { useLogger } from '../../utils/useLogger';
 import { getInitialState } from './gridState';
 
 export const useGridApi = (apiRef: ApiRef): GridApi => {
   const logger = useLogger('useGridApi');
+  const [, forceUpdate] = React.useState();
```

Comment:
```suggestion
  const [, forceUpdate] = React.useReducer((acc) => acc + 1, 0);
```

Reply:
Why? I prefer to useState so I can pass the state I need...

Reply:
Because it doesn't work?

Reply:
In this case, I find the logic confusion, what about calling setState?

Reply:
it does, you just need to pass a function

Reply:
> Why? I prefer to useState so I can pass the state I need...

The name `forceUpdate` is misleading. You have to pass the correct parameter to schedule work.

Though generally forcing a re-render is rarely needed. Whenever you `forceUpdate` why can't you move whatever changed into state?

Reply:
There is one case that took me some time to understand, I think that by doing 

```jsx
apiRef.current.forceUpdate(() => apiRef.current.state),
```

We can skip updates when the state hasn't changed, which can be an interesting feature. I was confused by the name, we definitely need to find a different one. IMHO `forceUpdate` === no argumetnts.

Reply:
this is not a setState, it's just to force the ui to rerender

Reply:
@dtassone Are you saying that all we need is a new reference? If its the case, I would rather do:

```diff
-apiRef.current.forceUpdate(() => apiRef.current.state),
+apiRef.current.forceUpdate({}),
```

to remove the dependency another part of the logic.

Reply:
> this is not a setState, it's just to force the ui to rerender

But it is? Assuming you refer to `setState` returned from the `useState` hook then I don't see any difference. Both `setState` and `forceUpdate` have to know what the current state is to actually force a re-rerender (which isn't clear why you need this).